### ISSUE TITLE
Seqencing assumes command argument `name` field is defined

### ISSUE THREAD

**Issue by cartermak** [NASA-AMMOS/aerie#907]
### Checked for duplicates

Yes - I've already checked

### Is this a regression?

No - This is a new bug

### Version

<=1.6.0

### Describe the bug

The SeqJSON schema states that the `name` field in a command argument is optional (for all argument types), however Aerie assumes in a couple of places that `name` is present. 

1. [In the EDSL preface](https://github.com/NASA-AMMOS/aerie/blob/51441ef0a5dfebeec4e0ed17b4935a980c4d32a3/sequencing-server/src/lib/codegen/CommandEDSLPreface.ts#L2311-L2371), `arg.name` is accessed without a typeguard for undefined values. This causes errors when downloading the EDSL TS library for command expansion or local sequence development (one of multiple such issues, already tracked here: #621). 

2. Parsing raw SeqJSON to EDSL also breaks here because it appears to rely on the name field corresponding to the command dictionary name. This conversion should use the argument order to assign argument values, not the argument name fields.

### Reproduction

1. Download the EDSL TS library file for any command dictionary and load it into an environment where Typescript linting is enabled.
2. Upload a valid sequence to the sequence editor where one or more command arguments do not have the `name` field defined. The editor will populate the name as `undefined` and then the linter will show an error.

### System Info

```shell
Observed in Aerie 1.5.0 and Aerie 1.6.0.
```

### Severity

Annoyance, blocker in limited cases

**Comment by cartermak** [comment]
To elaborate, this valid sequence using the Banananation dictionary:

```json
{
  "id": "test_seq",
  "metadata": {},
  "steps": [
    {
      "args": [
        {
          "type": "number",
          "value": 90
        }
      ],
      "stem": "PREHEAT_OVEN",
      "time": {
        "type": "COMMAND_COMPLETE"
      },
      "type": "command"
    }
  ]
}
```

becomes the following (incorrect) sequence when trying to upload to the Sequence Editor:

```ts
export default () =>
  Sequence.new({
    seqId: 'test_seq',
    metadata: {},
    steps: ({ locals, parameters }) => ([
      C.PREHEAT_OVEN({
        arg0: 90, // Note incorrect argument name
      }),
    ]),
  });
```

(Tested now in Aerie 1.10.0)

**Comment by joswig** [comment]
Fixed with #1093
